File: payment-ms/src/main/java/br/puc/tp_final/payment/PaymentController.java
package br.puc.tp_final.payment;

import jakarta.ejb.EJB;
import jakarta.ws.rs.*;

@Path("payment")
@Produces({"application/json"})
public class PaymentController {

    @EJB
    private PaymentService paymentService;

    @POST
    @Path("pay")
    @Consumes("application/json")
    public void pay() {
        paymentService.pay();
    }

    @GET
    @Path("/status/{id}")
    @Consumes("application/json")
    public String status() {
        return paymentService.status();
    }
}

File: payment-ms/src/main/kotlin/br/puc/tp_final/payment/PaymentController.kt
package br.puc.tp_final.payment

import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/payment-ms/rest/payment")
class PaymentController(
    val paymentService: PaymentService
) {
    @PostMapping("pay")
    fun pay() {
        paymentService.pay()
    }

    @GetMapping("status/{id}")
    fun status(): String {
        return paymentService.status()
    }
}